packages/protocol/scripts/sourcify-publish.ts

```diff
@@ -77,33 +61,66 @@ async function main(buildTargets: BuildOptions) {
     console.log('-'.repeat(artifact.contractName.length))
 
     const address = r.contract.includes('Proxy') ? r.args[0] : r.args[1]
+    const implementationName = r.contract.includes('Proxy')
+      ? artifact.contractName.replace('Proxy', '')
+      : artifact.contractName
+
     const formData = new FormData()
     formData.append(
       'files',
-      fs.createReadStream(artifactBasePath + '/' + artifact.contractName + '.json')
+      fs.createReadStream(artifactBasePath + '/' + implementationName + '.json')
     )
     formData.append('address', address)
     formData.append('chain', chainId)
 
-    await fetch('https://sourcify.dev/server', {
-      method: 'POST',
-      body: formData,
-    })
-      .then((res) => res.json())
-      .then((json) =>
-        fetch(
-          `https://${network}-blockscout.celo-testnet.org/address/${json.result[0].address}/contracts`
-        )
-      )
+    const mainContract = artifacts.find((a) => a.contractName === implementationName)
+    const sourcesfilesNeeded = Object.keys(JSON.parse(mainContract.metadata).sources)
+
+    const p = sourcesfilesNeeded.find((f) => f.includes(implementationName))
+    // Interface contract needs the parent implementation as well
```

Comment:
What do you mean by interface contract?

Reply:
I mean that is Single Inheritance from a parent contract

Reply:
Interesting that this is necessary here. Why is this not necessary for other contracts with inheritance to verify their full bytecode?